aemedge/blocks/contract-spec/contract-spec.js

```diff
@@ -0,0 +1,766 @@
+      // MinimumPriceFluctuation format
+      return value.map((item) => item.mintk).join('<br />');
+    }
+    if (value[0].contrMonth) {
+      // ListedContracts format
+      return value.map((item) => item.contrMonth).join('<br />');
+    }
+    return value.join('<br />');
+  }
+
+  // Handle objects with nested structure
+  if (typeof value === 'object') {
+    return JSON.stringify(value);
+  }
+
+  return String(value);
+}
+
+/**
+ * Format field name for display (convert camelCase to Title Case)
+ * @param {string} fieldName - The camelCase field name from API
+ * @returns {string} Formatted field name (e.g., "Contract Unit")
+ */
+function formatFieldName(fieldName) {
+  return fieldName
+    .replace(/([A-Z])/g, ' $1')
+    .replace(/^./, (str) => str.toUpperCase())
+    .trim();
+}
+
+/**
+ * Get tooltip text for a given field name
+ * Returns the tooltip key for i18n translation
+ * @param {string} fieldName - The field name to get tooltip for
+ * @returns {string} Tooltip text key for i18n
+ */
+function getTooltipText(fieldName) {
+  const tooltips = {
+    ContractUnit: 'The contract unit is the quantity of the product delivered for a single contract.',
+    PriceQuotation: 'The price quotation is the contract amount expressed in currency (e.g. dollars and cents) per unit of the product (e.g. per pound, per bushel, etc.).',
+    ProductCode: 'The product code is a one- to three-letter code identifying the product, followed by additional characters indicating the month and year of expiration.',
+    TradingHours: 'The trading hours are the hours of operation for trading a product.',
+  };
+  return tooltips[fieldName] || `Information about ${formatFieldName(fieldName).toLowerCase()}.`;
+}
+
+/**
+ * LOCAL DEV FALLBACK - TODO: Remove this function before production
+ * Fetches mock contract specs data from local JSON file for localhost development
+ * @returns {Promise<Object|null>} Mock contract specs data or null
+ */
+async function fetchContractSpecsLocalDev() {
+  try {
+    const response = await fetch('/aemedge/blocks/contract-spec/300.json');
+    if (response.ok) {
+      const data = await response.json();
+      devLog('Using local dev contract specs data from 300.json');
+      return data;
+    }
+    if (IS_LOCALHOST) {
+      // eslint-disable-next-line no-console
+      console.warn('Local dev file not found, falling back to API');
+    }
+  } catch (e) {
+    if (IS_LOCALHOST) {
+      // eslint-disable-next-line no-console
+      console.error('Error fetching local dev contract specs:', e);
+    }
+  }
+  return null;
+}
+
+/**
+ * Fetch contract specs from API or use local dev fallback
+ * @param {string|number} productId - The product ID to fetch specs for
+ * @param {string} apiEndpoint - The API endpoint path (configurable)
+ * @returns {Promise<Object|null>} Contract specs data or null on error
+ */
+async function fetchContractSpecs(productId, apiEndpoint = DEFAULT_API_ENDPOINT) {
```

Comment:
We put all API calls at product util file, please refer to `getContractsByNumber` function